public/app/features/transformers/timeSeriesTable/timeSeriesTableTransformer.ts

```diff
@@ -172,7 +172,7 @@ export function timeSeriesToTableTransform(options: TimeSeriesTableTransformerOp
         // and push the frame with reduction
         // into the the appropriate field
         const reducerId = options[refId]?.stat ?? ReducerID.lastNotNull;
-        const value = reduceField({ field, reducers: [reducerId] })[reducerId] || null;
```

Comment:
You're correct in that this check should not be `|| null`. But looking at the implementation of `reduceField` it seems to return undefined if the field has no values and [`reducer.emptyInputResult`](https://github.com/grafana/grafana/blob/b607a4e0a2ee2a9b368c6ebcf4021249768b347e/packages/grafana-data/src/transformations/fieldReducer.ts#L90C19-L90C19) is undefined. It could be that the check in reduceField is wrong, and it should really check if emptyInputResult is undefined.

In any case, I think the change here should be to check if the FieldCalc here is `undefined`. Adding a test with a field that has no values would also be a good idea.